src/hypothesis/extra/datetime.py

```diff
@@ -123,3 +123,34 @@ def times(allow_naive=None, timezones=None):
 
 def datetime_to_time(dt):
     return dt.timetz()
+
+
+class TimedeltaStrategy(SearchStrategy):
+
+    def __init__(self, min_value=dt.timedelta.min, max_value=dt.timedelta.max):
+        assert type(min_value) == dt.timedelta, 'min_value must be a timedelta'
```

Comment:
Also, rather than using assertions, we should be raising an `InvalidArgument` exception.

(As with other comments, I recommend looking at `strategies.py` for inspiration.)